code/datums/gamemode/objectives/target/assassinate.dm

```diff
@@ -99,25 +99,26 @@ var/list/assassination_objectives = list()
 		var/datum/component/uplink/enemy_uplink = target.find_syndicate_uplink(enemy.uplink)
 		//chances are the target's uplink is no longer on their mind.current especially if they got decapitated or such.
 		//by associating the uplink with the role we can at least try and get the TCs out of it.
+		//Increases the victor's uplink (if it finds it) by 8tcs, or the number in the victim's uplink, whichever was greater, and sets the victim's to 0 (if it finds it).
 
 		if (A.target == owner)
-			to_chat(owner.current, "<span class='notice'>The Syndicate congratulates you on your Victory. Look forward to be assigned on higher risk operations another day.</span>")
+			to_chat(owner.current, "<span class='notice'>The Syndicate congratulates you on your victory. Look forward to be assigned on higher risk operations another day.</span>")
 		else
 			if (owner_uplink)
-				owner_uplink.telecrystals += DOUBLE_AGENT_TC_REWARD
-				if (enemy_uplink)
-					owner_uplink.telecrystals += enemy_uplink.telecrystals
-					enemy_uplink.telecrystals = 0
-				to_chat(owner.current, "<span class='notice'>Good work agent. [DOUBLE_AGENT_TC_REWARD] additional tele-crystals have been sent to your uplink.</span>")
+				owner_uplink.telecrystals += max(reward, enemy_uplink?.telecrystals)
+				to_chat(owner.current, "<span class='notice'>Good work, agent. 8 plus any remaining additional tele-crystals over 8 from the target's uplink have been sent to your uplink.</span>")
 			else
-				to_chat(owner.current, "<span class='notice'>Good work agent. Unfortunately we couldn't find your uplink on your person, so no additional tele-crystals could be distributed.</span>")
+				to_chat(owner.current, "<span class='notice'>Good work, agent. Unfortunately, we couldn't find your uplink on your person, so no additional tele-crystals could be distributed.</span>")
+			if (enemy_uplink)
+				enemy_uplink.telecrystals = 0
```

Comment:
why are you setting it to 0 up here when you do it below?

won't this cause their telecrystals to become 0 BEFORE the add and then always default to 8 TC for the assassin?

Reply:
For some reason the original code did it somewhat weirdly where it had the TC-editing clause within the if-statement that checks if there are any other challengers to become a new target or not, so they had to have a separate TC editing clause after that statement for the situation where there was indeed a new target, since the previous one wouldn't be executed in that case. I'm gonna try making it cleaner and clearer by separating them since they don't really have any reason to be set up that way (that I can tell).

Reply:
Okay please do, and add comments

Reply:
Okay, I think this last commit should do it.